src/program.rs

```diff
@@ -406,6 +406,12 @@ pub(crate) enum BufferInitContent {
     },
 }
 
+#[derive(Debug)]
+pub(crate) struct DeferredBufferInitContentBuilder<'trgt> {
+    buf: &'trgt mut Vec<u8>,
+    start: usize,
+}
+
```

Comment:
Thanks, I hate it :joy: But yeah, seriously the `new` should be `build` method in `BufferInitContent` instead because there's no reason one should have to `use` this directly. (Not an extension trait because I want to switch `Vec<u8>` to a custom type soonish).

Reply:
Also just `BufferInitContentBuilder`. There is no deferring going on, you can't unborrow the vector until a later date to resume.